### DISCUSSION TITLE
Access instance inside a job

### DISCUSSION THREAD

**Discussion by abetoots** [breejs/bree#182] [Ideas] [answered]
How do I access the instance inside a job? Ideally, I would want to do a `bree.add()` inside a running polling job. This way, my polling job is handled by Bree and inside the polling job, I could schedule jobs to be also handled by bree. Is this possible? Or should I just do the polling separately instead of it being a bree job?

**Comment by shadowgate15** [comment, marked as answer]
I would probably use the [`workerMessageHandler`](https://github.com/breejs/bree#custom-errormessage-handling). This would then allow the job to send a message like `{ event: 'add', data: JOB_CONFIG}` and then in the `workerMessageHandler` function you could call `bree.add()`.

**Reply by abetoots** [reply]
that works. thanks for pointing me in the right direction! so instead of setting `workerMessageHandler` during instantiation, i'm setting it through `bree.config.workerMessageHandler((message, workerMetadata) => { bree.add() })`
